src/JsRoot/JsRootGraph2D.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from 'react';
+import Skeleton from '@mui/material/Skeleton';
+import { useJSROOT } from './JsRootService';
+import { useVisible } from 'react-hooks-visible';
+import { Page2D } from './GraphData';
+
+export function JsRootGraph2D(props: Page2D) {
+	const { JSROOT } = useJSROOT();
+	const [containerEl, visible] = useVisible<HTMLDivElement>();
+
+	const [obj, setObj] = useState(undefined);
+	const [drawn, setDrawn] = useState(false);
+	const [isVisible, setIsVisible] = useState(false);
+
+	useEffect(() => {
+		setIsVisible(visible > 0.3);
+		return () => setIsVisible(false);
+	}, [visible]);
+
+	useEffect(() => {
+		if (drawn || !visible) return;
+        const x = props.first_axis.values;
+        const y = props.second_axis.values;
+        const z = props.data.values
+		
+		// JSRoot is unable to draw bins with value from the end of the range unless there is at least
```

Comment:
It is not a problem and rather a feature, I am quite happy with white bins for which value is zero.